b2/console_tool.py

```diff
@@ -420,21 +422,42 @@ def run(self, args):
 
 class GetBucket(Command):
     """
-    b2 get-bucket <bucketName>
+    b2 get-bucket [--showSize] <bucketName>
 
         Prints all of the information about the bucket, including
         bucket info, CORS rules and lifecycle rules.
+
+        If --showSize is specified, then display the number of files
+        (fileCount) in the bucket and the aggregate size of all files
+        (totalSize).
     """
 
+    OPTION_FLAGS = ['showSize']
+
     REQUIRED = ['bucketName']
 
     def run(self, args):
         # This always wants up-to-date info, so it does not use
         # the bucket cache.
         for b in self.api.list_buckets():
             if b.name == args.bucketName:
-                self._print(json.dumps(b.bucket_dict, indent=4, sort_keys=True))
-                return 0
+                if not args.showSize:
+                    self._print(json.dumps(b.bucket_dict, indent=4, sort_keys=True))
+                    return 0
+                else:
+                    files = list(b.ls("", args.bucketName))
```

Comment:
For large buckets, it would be better not to build the entire list of files.  `ls` yields the results one at a time.  If you process them one at a time without building a list, it won't require so much memory.

Right now, there are buckets in B2 with almost a billion files.

Reply:
I was wondering about this while I was coding it. I recognized that it wouldn't scale for very large buckets, but didn't have much idea about the scalability of the system as a whole in any case. I'll rewrite to accumulate without interpolating the entire generator.